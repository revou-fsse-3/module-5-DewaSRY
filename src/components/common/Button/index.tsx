import { FC, ButtonHTMLAttributes, ReactNode, PropsWithChildren } from "react";
export const enum ButtonTypes {
  BaseButton = "border-none ",
  PrimaryButton = "bg-green-400 text-white border-green-400 ",
  SecondaryButton = "bg-blue-400 text-white border-blue-400 ",
  TernaryButton = "bg-sky-400 text-white border-sky-400 ",
  FourButton = "bg-red-400 text-white border-red-400 ",
  FiveButton = "bg-yellow-400 text-white border-yellow-400 ",
}

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  ButtonType?: ButtonTypes;
  hidedButton?: boolean;
  activeButton?: boolean;
  isLoading?: boolean;
}
type ButtonComponent = FC<ButtonProps> & PropsWithChildren;
const Button: ButtonComponent = ({
  children,
  ButtonType = "",
  hidedButton = false,
  activeButton = false,
  isLoading = false,
  ...restProps
}) => {
  return (
    <button
      {...restProps}
      disabled={isLoading}
      className={
        `${hidedButton ? " hidden " : " inline-block "}` +
        `${activeButton ? " brightness-75 text-black " : "  "}` +
        " my-1 border-2 px-8 text-lg rounded-sm " +
        `${restProps.disabled ? "" : "hover:brightness-50 "} ` +
        " disabled:brightness-10 disabled:border-none  disabled:bg-gray-600 " +
        ButtonType +
        `${restProps.className ? restProps.className : ""}`
      }
    >
      {children}
    </button>
  );
};

export default Button;
