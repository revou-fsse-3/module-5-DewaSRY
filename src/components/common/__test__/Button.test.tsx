import { describe, expect, vi, it, beforeAll, beforeEach } from "vitest";
import { render, fireEvent, screen, act, waitFor } from "@libs/reactTesting";
import "@testing-library/jest-dom";
import Button from "@common/Button";

const mocks = vi.hoisted(() => {
  return {
    handle: vi.fn(),
  };
});

describe("testing Table Category container", () => {
  beforeEach(() => {
    vi.clearAllMocks();
    render(<Button role="main"> Button </Button>);
  });

  it("table Category match with snapshot ", () => {
    const main = screen.getAllByRole("main");
    expect(main).toMatchSnapshot();
  });
});
