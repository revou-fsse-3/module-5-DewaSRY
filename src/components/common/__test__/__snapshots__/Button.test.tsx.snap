// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing Table Category container > table Category match with snapshot  1`] = `
[
  <button
    class=" inline-block    my-1 border-2 px-8 text-lg rounded-sm hover:brightness-50   disabled:brightness-10 disabled:border-none  disabled:bg-gray-600 "
    role="main"
  >
     Button 
  </button>,
]
`;
